sdk/metric/labelencoder.go

```diff
@@ -54,9 +61,27 @@ func (d *defaultLabelEncoder) Encode(labels []core.KeyValue) string {
 		if i > 0 {
 			_, _ = buf.WriteRune(',')
 		}
-		_, _ = buf.WriteString(string(kv.Key))
+		copyAndEscape(buf, string(kv.Key))
+
 		_, _ = buf.WriteRune('=')
-		_, _ = buf.WriteString(kv.Value.Emit())
+
+		if kv.Value.Type() == core.STRING {
+			copyAndEscape(buf, kv.Value.AsString())
+		} else {
+			_, _ = buf.WriteString(kv.Value.Emit())
+		}
 	}
 	return buf.String()
 }
+
+func copyAndEscape(buf *bytes.Buffer, val string) {
+	for _, ch := range val {
+		switch ch {
+		case '=', ',', escapeChar:
+			buf.WriteRune(escapeChar)
+			fallthrough
+		default:
+			buf.WriteRune(ch)
+		}
```

Comment:
Would you mind to just move to outside `buf.WriteRune(ch)` to outside of the switch? `fallthrough` and `default` statements just makes it harder to read.

```suggestion
		switch ch {
		case '=', ',', escapeChar:
			buf.WriteRune(escapeChar)
		}
		buf.WriteRune(ch)
```

Reply:
Done. Not sure what I was thinking there. :)